dio/lib/src/sentry_transformer.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:dio/dio.dart';
+import 'package:sentry/sentry.dart';
+
+/// A transformer which wraps transforming in spans
+class SentryTransformer implements Transformer {
+  // ignore: public_member_api_docs
+  SentryTransformer({required Transformer transformer, Hub? hub})
+      : _hub = hub ?? HubAdapter(),
+        _transformer = transformer;
+
+  final Transformer _transformer;
+  final Hub _hub;
+
+  @override
+  Future<String> transformRequest(RequestOptions options) async {
+    final span = _hub.getSpan()?.startChild(
+          'transform-request',
```

Comment:
According to https://develop.sentry.dev/sdk/performance/span-operations/ this should probably be `serialize`?